app/models/pet.rb

```diff
@@ -10,4 +13,16 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def self.search_by_name(name)
+    where("name ILIKE ?", "%#{name}%")
```

Comment:
Check out the ApplicationRecord -- there is a search function already built for you there that you can use! It is inherited into all models because all models inherit from ApplicationRecord. That means, you could call `Pet.search(search_term)` without implementing search again.